simplified-cardcreator/src/main/java/org/nypl/simplified/cardcreator/ui/LocationFragment.kt

```diff
@@ -0,0 +1,263 @@
+
+  private val locationRequestCode = 102
+
+  // Minimum distance between location updates, in meters
+  private val minDistanceUpdates: Float = 10f
+
+  // Minimum time interval between location updates, in milliseconds
+  private val minTimeUpdates = 1000 * 60.toLong()
+
+  override fun onCreateView(
+    inflater: LayoutInflater,
+    container: ViewGroup?,
+    savedInstanceState: Bundle?
+  ): View? {
+    _binding = FragmentLocationBinding.inflate(inflater, container, false)
+    return binding.root
+  }
+
+  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+    super.onViewCreated(view, savedInstanceState)
+
+    navController = Navigation.findNavController(activity!!, R.id.card_creator_nav_host_fragment)
+    isInNewYorkState()
+
+    binding.checkLocationBtn.setOnClickListener {
+      isInNewYorkState()
+    }
+
+    // Go to next screen
+    binding.nextBtn.setOnClickListener {
+      if (isNewYork) {
+        logger.debug("User navigated to the next screen")
+        nextAction = LocationFragmentDirections.actionNext()
+        navController.navigate(nextAction)
+      } else {
+        val data = Intent()
+        activity!!.setResult(Activity.RESULT_CANCELED, data)
+        activity!!.finish()
+      }
+    }
+
+    // Go to previous screen
+    binding.prevBtn.setOnClickListener {
+      activity!!.onBackPressed()
+    }
+  }
+
+  /**
+   * Show loading screen
+   */
+  private fun showLoading(show: Boolean) {
+    logger.debug("Toggling loading screen")
+    if (show) {
+      binding.loading.visibility = View.VISIBLE
+    } else {
+      binding.loading.visibility = View.GONE
+    }
+  }
+
+  /**
+   * Checks to see if the user has granted required location permissions
+   */
+  private fun getLocation(): Location? {
+    var location: Location? = null
+    logger.debug("Checking for location permission")
+    if (ContextCompat.checkSelfPermission(activity!!, Manifest.permission.ACCESS_COARSE_LOCATION)
+      != PackageManager.PERMISSION_GRANTED
+    ) {
+      logger.debug("Requesting location permission")
+      ActivityCompat.requestPermissions(
+        activity!!,
+        arrayOf(Manifest.permission.ACCESS_COARSE_LOCATION),
+        locationRequestCode
+      )
+    } else {
+      logger.debug("Location permission granted")
+      try {
+        logger.debug("Getting current location")
+        val locationManager = activity!!.getSystemService(Context.LOCATION_SERVICE) as LocationManager
+        val isNetworkLocationEnabled = locationManager.isProviderEnabled(LocationManager.NETWORK_PROVIDER)
+
+        if (isNetworkLocationEnabled) {
+          locationManager.requestLocationUpdates(LocationManager.NETWORK_PROVIDER, minTimeUpdates, minDistanceUpdates, this)
+          location = locationManager.getLastKnownLocation(LocationManager.NETWORK_PROVIDER)
+        }
+
+      } catch (e: Exception) {
+        logger.debug("Error getting current location")
+        e.printStackTrace()
```

Comment:
See notes below about `printStackTrace`

Reply:
Is there something better we can do if this fails? The previous card creator had this exact problem with essentially swallowing errors...

Reply:
@io7m We could show a dialog here and direct the user to their device location settings here

Reply:
Is this required? What if they don't want to give us their location? Is there a manual work around?

Reply:
We might need to loop @killerpuppytails in on this one. I would suppose we can't let them sign up for the card if we can't without a doubt determine whether or not they are in New York at the time of creating the card. But since they can have a work/school address in New York...maybe they are on vacation and what to sign up for SimplyE....with our current flow, they can't do that.

Reply:
Yes, their location is required, and they MUST be in the confines of NYS when they apply. It's the only way we can truly confirm they satisfy the requirements of card creation (they must live in, work in, or be visiting NYS) without them coming in to a branch or uploading ID (which we want to avoid).

Reply:
Got it! Thanks y'all.